### ISSUE TITLE
Accept privacy terms: checkboxes unusable for screen reader users

### ISSUE THREAD

**Issue by zersiax** [freeCodeCamp/freeCodeCamp#17730]
#### Describe your problem and - if possible - how to reproduce it
Currently, the checkboxes to accept the privacy terms, the newsletter opt-in etc. are completely inaccessible to screen reader users. They do not convey a role of checkbox, are not keyboard-focusable and do not respond to regular keypresses (e.g. space, enter) to check them.  This effectively halts any screen reader user at the door, since at first sight there is no way around this prompt.

#### Add a Link to the page with the problem
https://www.freecodecamp.org/accept-privacy-terms

#### Tell us about your browser and operating system
Chrome latest stable, on Windows 10 1803
* Browser Name: 
* Browser Version: 
* Operating System: 

#### Proposed patch:
The first rule of ARIA is to not use ARIA, since HTML semantics can often fix the problems ARIA is often used to fix. Best case scenario, native HTML checkbox elements are modified to perform the same function the non-standard HTML is currently performing. 
Since these checkboxes appear to deviate from standards quite a bit, I haven't looked at the code yet, adding a 'role=checkbox' to the relevant HTML elements will at least make sure stuff works for now. I stress though that this is a hack, using ARIA to simulate HTML native behavior is a pretty big antipattern.

**Comment by zersiax** [comment]
For now, if anyone else runs into this, maximize the browser window and use NVDA's OCR addon to snap a clear picture of the page, use NVDA's review cursor to find the start of the lines containing the checkboxes than simulate a left mouse click, that seems the only way to get past it until a fix has been implemented.  Haven't tested with JAWS and I'm pretty sure this trick won't work with Narrator, VoiceOver on Mac OS can probably use the VOCR package to perform the same trick

**Comment by QuincyLarson** [comment]
Thanks for letting us know, @zersiax. I hadn't realized that checkboxes were an accessibility anti-pattern.

We're working on a solution that won't involve checkboxes.

I've renamed this issue to focus on the checkbox-related HTML challenges in our curriculum. We should definitely update that so that it's in line with accessibility best practices.

**Comment by zersiax** [comment]
Honestly, if you are referring to challenge https://learn.freecodecamp.org/responsive-web-design/basic-html-and-html5/create-a-set-of-checkboxes , that is the exact way checkboxes should be used. That, however, is not the approach that was taken on the accept privacy terms page above.

**Comment by zersiax** [comment]
The checkboxes on said page are more along the lines of this:
```html
<div class="checkbox">
<label><input id="terms" name="privacy" type="checkbox">
<span class="cr"><i class="cr-icon fa fa-check"></i></span> I accept the <a href="https://www.freecodecamp.org/terms" target="_blank">terms of service</a> (required).
```
etc.
I'd need to do testing why that isn't working though, there is a checkbox element in there but something is muddying up the waters.

**Comment by zersiax** [comment]
After some limited testing, I can say with reasonable certain that the HTML markup is not the culprit here. I tried this exact code in an otherwise empty file, without adding the required CSS and JS files. This makes the checkbox work just fine, so it's likely one of these that's causing problems.

**Comment by QuincyLarson** [comment]
OK - so I talked with @zersiax and it is indeed this page and the way the checkboxes have been implemented that cause accessibility issues. Here's the code we're currently using, which is suboptimal:

```
<div class="checkbox"><label><input id="terms" name="privacy" type="checkbox"><span class="cr"><i class="cr-icon fa fa-check"></i></span> I accept the <a href="https://www.freecodecamp.org/terms " target="_blank">terms of service</a> (required) ..... </div>
```

**Comment by raisedadead** [comment]
We are removing the need for explicit acceptance of these privacy terms. Please see linked issue for details. This view can be removed.
